### PR TITLE
Type Mismatch in Ajax table

### PR DESCRIPTION
**Pull request by LouisMuriuki**
There is a type mismatch between the onChange prop in the Table component and the handleTableChange function passed to it. The onChange prop is expecting a function with the signature: `(pagination: TablePaginationConfig, filters: Record<string, FilterValue | null>, sorter: SorterResult<DataType> | SorterResult<...>[], extra: TableCurrentDataSource<...>) => void But handleTableChange has the signature:
`(pagination: TablePaginationConfig, filters: Record<string, FilterValue>, sorter: SorterResult<DataType>) => void `
A seen null value is not accounted for

Again the sorter parameter does not match the type declared in the function signature `sorter as SorterResult<DataType>`, but your implementation is defining it as `SorterResult<DataType> | SorterResult<DataType>[]`

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
Could you provide a reproduce?

**Comment by afc163** [comment]
> Could you provide a reproduce?

ping

**Comment by LouisMuriuki** [comment]
Hello, i will check if the error exists and try to reproduce it

**Comment by LouisMuriuki** [comment]
Hello @afc163 , sorry for the late reply but here is the reproducible example https://github.com/LouisMuriuki/AjaxReproducible
it does not necessarily crash the application but TypeScript complains as such
`Type '(pagination: TablePaginationConfig, filters: Record<string, FilterValue>, sorter: SorterResult<DataType>) => void' is not assignable to type '(pagination: TablePaginationConfig, filters: Record<string, FilterValue | null>, sorter: SorterResult<DataType> | SorterResult<...>[], extra: TableCurrentDataSource<...>) => void'.
  Types of parameters 'filters' and 'filters' are incompatible.
    Type 'Record<string, FilterValue | null>' is not assignable to type 'Record<string, FilterValue>'.
      'string' index signatures are incompatible.
        Type 'FilterValue | null' is not assignable to type 'FilterValue'.
          Type 'null' is not assignable to type '(boolean | Key)[]'.ts(2322)
InternalTable.d.ts(21, 5): The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & TableProps<DataType> & { children?: ReactNode; } & { ref?: Ref<HTMLDivElement> | undefined; }`
My changes fixxes the problem.
Reach out if you need more information

File: components/table/demo/ajax.tsx [hunk 1]
```diff
@@ -87,12 +87,12 @@ const App: React.FC = () => {
 
   const handleTableChange = (
     pagination: TablePaginationConfig,
-    filters: Record<string, FilterValue>,
-    sorter: SorterResult<DataType>,
+    filters: Record<string, FilterValue | null>,
+    sorter: SorterResult<DataType> | SorterResult<DataType>[]
```
Comment (afc163) [inline components/table/demo/ajax.tsx:91]:
```suggestion
    sorter: SorterResult<DataType> | SorterResult<DataType>[],
```

**Review by luihugo — APPROVED** [review]

File: components/table/demo/ajax.tsx (hunk 1 above)
Comment (afc163) [reply components/table/demo/ajax.tsx:91]: ping~

**Pull request closed without merging**